pygfx/utils/compgeo.py

```diff
@@ -145,3 +141,24 @@ def value_from_t(t):
         t = samples_t[1] + dt * t_step
 
     return t
+
+
+# We use our own bisect functions. Otherwise, to use bisect.bisect with descending
+# values, we need the key param, which adds complexity, but also does not work pre py10.
+# This solution even seems a wee bit faster than using builtin bisect.
+
+
+def bisect_asc(values, ref):
+    """Simple version of bisect.bisect(). Values are assumed to be in ascending order."""
+    for i, v in enumerate(values):
+        if v > ref:
+            return i
+    return i + 1
+
+
+def bisect_desc(values, ref):
+    """Like bisect(), but the values are assumed in descnding order."""
+    for i, v in enumerate(values):
+        if v <= ref:
+            return i
+    return i + 1
```

Comment:
Bisect builtin is supposed to be very fast because it uses binary search and is implemented in C I think. It's surprising to see that your naive for loop performs better? I can only imagine it to be true if you happen to be testing on examples where the lists are very small or the bisect value is always an early value.

Reply:
Our lists are always just 3 items.

We don't have a big list to find the insertion point form but more of a description of a line of which we need to find the reference point. We call bisect repeatedly as part of the binary search.

Reply:
So I guess it's not so much the bisect function itself that helped improve this code, but more the idea of bisection :)